test/complex_test.exs

```diff
@@ -0,0 +1,53 @@
+defmodule ComplexTest do
+  use ExUnit.Case
+  # Calculated by trial and error :(
+  @epsilon 1.0e-15
+
+  test "can be created with numbers representing real and imaginary parts" do
+    z = Complex.new(1, 2.0)
+    assert z.re == 1
+    assert z.im == 2
+  end
+
+  test "can be created from a 2-tuple of real and imaginary parts" do
+    assert Complex.new(1, 2) == Complex.new({1, 2})
```

Comment:
If you want to be really liberal about what you accept, you can add additional function bodies to the other functions that pipe any input through new and then into themself again.

Like so:

```elixir
def subtract(a = %__MODULE__{}, b = %__MODULE__{}) do
  new(a.re - b.re, a.im - b.im)
end
def subtract(a, b), do: subtract(new(a), new(b)  # we kind of assume we're getting tuples here...
```

This probably makes you want to add additional function bodies for `new` so that it can handle the argument already being a `Complex` struct (effectively a no-op).

Personally I'd be more than fine having to provide `Complex` structs to the various functions. The `new` that does a no-op on an `Complex`struct would be a nice addition though.